test/reload_evolution-luatest/reload_router_test.lua

```diff
@@ -0,0 +1,277 @@
+local t = require('luatest')
+local vtest = require('test.luatest_helpers.vtest')
+local os = require('os')
+
+-- Legacy utils
+local lutil = require('test.lua_libs.util')
+local lgit_util = require('test.lua_libs.git_util')
+
+local g = t.group('reload_router')
+local cfg_template = {
+    sharding = {
+        {
+            replicas = {
+                replica_1_a = {
+                    master = true,
+                },
+                replica_1_b = {},
+            },
+        },
+        {
+            replicas = {
+                replica_2_a = {
+                    master = true,
+                },
+            },
+        },
+    },
+    bucket_count = 100
+}
+
+local global_cfg
+
+g.before_all(function()
+    global_cfg = vtest.config_new(cfg_template)
+
+    -- We cannot work at the same directory, where the test is.
+    rawset(_G, 'vshard_copy_path', lutil.BUILDDIR ..
+           '/test/var/vshard_git_tree_copy_router_luatest')
```

Comment:
Now `VARDIR` is used

Reply:
Are you sure? I don't see it being used in the patch.

Reply:
Temporary directories are used now. They're unique enough to allow running test in parallel. See `before_all`:
```lua
    g.vshard_copy_path = fio.tempdir()
    -- Copy source to the temporary directory
    t.assert_equals(fio.mkdir(g.vshard_copy_path .. '/.git'), true)
    t.assert_equals(fio.copytree(vtest.sourcedir .. '/.git',
        g.vshard_copy_path .. '/.git'), true)
```

However, new test-run already uses temporary directories by default, and creating new one just for that test don't make a lot of sense. So, let's allow user or test-run to decide, where to put our files and use `fio.tempdir` only as fallback option, if no `VARDIR` is provided.

Reply:
In the new version you will call `fio.tempdir()` on each load of `vtest` when `VARDIR` is not specified. Even when `vtest.vardir` is not needed. Which looks like an overkill, because 1) the non-main instances won't have `VARDIR`, 2) `fio.tempdir()` actually creates a directory. So after running all luatests you will have N temp directories created where N - number of storages and routers started and restarted ever. Even though you needed it just once for a single test.

I would suggest to do that just in the new test file. Not in `vtest.lua`.